### DISCUSSION TITLE
regex to the file pattern glob

### DISCUSSION THREAD

**Discussion by lucky3582** [BurntSushi/ripgrep#2008] [Q&A] [answered]
Hi,
I am trying to use the regex to search in only particular files. 

I am trying to search only in files abc.<1 or 2 digit number>
so this should match 
abc.1
abc.23
abc.3

but not match
abc.23f
abc.444

I tried this regex but it none of this worked
abc.[0-9].?
abc.[0-9]{1,2}
abc.\d{1,2}

is there a way we can achieve this?

**Comment by BurntSushi** [comment, marked as answer]
When asking questions, it is always helpful to include the actual commands you're running, along with enough info to reproduce what you're seeing. This way, we can rule out other possible problems. For example, ripgrep doesn't have any option to filter files by regexes. So the fact that you're using regexes to filter files doesn't really make any sense to me.

If you want to use a regex, then I'd combine it with xargs:

```
rg --files -0 | rg '/abc.[0-9]{1,2}$' --null-data | xargs -0 rg <pattern>
```

If your file paths don't have whitespace in them, then it could be simpler:

```
rg --files | rg '/abc.[0-9]{1,2}$' | xargs rg <pattern>
```

**Reply by lucky3582** [reply]
sorry i am using this command with -g
rg -g "abc.[0-9]{1,2}" "help" ./

this is matching files having 2 digits that are only 1 or 2 
output
abc.11
abc.12
abc.21

**Reply by BurntSushi** [reply]
Yes, `-g` accepts globs, not regexes.

**Reply by lucky3582** [reply]
oh ok. thanks.

**Reply by lucky3582** [reply]
I tried this in  glob pattern finder which works but the same thing doesn't work when given for -g?
https://www.digitalocean.com/community/tools/glob?comments=true&glob=abc.%3F%28%5B0-9%5D%29%5B0-9%5D&matches=false&tests=abc.08&tests=abc.1&tests=abc.13f

rg -g "abc.?([0-9])[0-9]" "help" ./

rg -g "abc.?\([0-9]\)[0-9]" "help" ./

**Reply by BurntSushi** [reply]
"extended" glob syntax isn't supported by ripgrep.

**Reply by BurntSushi** [reply]
Glob implementations usually differ, unless there is some kind of conformance guarantee provided (like being POSIX compatible). So just because you can use an advanced feature somewhere doesn't mean you can use it everywhere.

**Reply by lucky3582** [reply]
ok thanks for the info and response :)
